File: Softeer/강의실 배정.py
from sys import stdin
n = int(stdin.readline())
lectures = []
for _ in range(n):
    s, f = map(int, stdin.readline().split())
    lectures.append([s, f])

lectures.sort(key = lambda x : x[1])
cnt = 1
last = lectures[0][1]

for i in range(1, n):
    if last <= lectures[i][0]:
        cnt += 1
        last = lectures[i][1]
    else:
        continue

print(cnt)

#---------------------------------------
from sys import stdin
import heapq
n = int(stdin.readline())
lectures = []
for _ in range(n):
    s, f = map(int, stdin.readline().split())
    heapq.heappush(lectures, (f, s))

cnt = 1
last = heapq.heappop(lectures)[0]

while lectures:
    now = heapq.heappop(lectures)
    if last <= now[1]:
        cnt += 1
        last = now[0]
    else:
        continue
print(cnt)

File: Softeer/강의실 배정.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const n = parseInt(input[0]);
const lectures = [];

for (let i = 1; i <= n; i++) {
    const [s, f] = input[i].split(' ').map(Number);
    lectures.push([s, f]);
}

lectures.sort((a, b) => a[1] - b[1]);

let cnt = 1;
let last = lectures[0][1];

for (let i = 1; i < lectures.length; i++) {
    const now = lectures[i];
    if (last <= now[0]) {
        cnt += 1;
        last = now[1];
    }
}
console.log(cnt);